packages/adapters/ethers/src/client.ts

```diff
@@ -868,16 +862,19 @@ export class EthersAdapter {
     if (address) {
       if (isEipNetwork) {
         this.appKit?.setPreferredAccountType(preferredAccountType, this.chainNamespace)
-        this.appKit?.setCaipAddress(`${caipNetworkId}:${address}`, this.chainNamespace)
+        this.appKit?.setCaipAddress(
+          `${this.chainNamespace}:${caipNetworkId}:${address}`,
```

Comment:
you sure this doesn't break anything? On solana I believe we receive a `CaipNetworkId`as response, causing this to be like eip155:solana:andsasdad:...